server/database/fake_data/progress.js

```diff
@@ -0,0 +1,56 @@
+const Progress = require('../models/progress');
+const User = require('../models/user');
+const Goal = require('../models/goal');
+
+const goalId = async (goalDescription) => {
+  const goal = await Goal.findOne({ description: goalDescription });
+  return goal.id;
+};
+const userId = async (handler) => {
+  const user = await User.findOne({ githubHandler: handler });
+  return user.id;
+};
+
+module.exports = () => new Promise(async (resolve) => {
```

Comment:
I don't see the need to use Promise here since ``` Progress.insertMany(progressArray);``` will give you what you need from the promise though it doesn't return a Promise  [check this](https://mongoosejs.com/docs/queries.html#queries-are-not-promises)
also what I think happens here that you'll have two `await` in a row,  here in line 54 and in index.js in line23

maybe we shouldn't change it for this PR, but we should consider it for next tasks

Reply:
nvm we've been awaiting them all day lol. But mongoose docs say it's not a true promise, so it's a bit confusing.

Reply:
as I said the code feels a little messy. Let's go over it in the refactor days.